src/tests/ftest/osa/osa_offline_drain.py

```diff
@@ -22,27 +23,30 @@ def setUp(self):
         """Set up for test case."""
         super(OSAOfflineDrain, self).setUp()
         self.dmg_command = self.get_dmg_command()
-        self.ior_apis = self.params.get("ior_api", '/run/ior/iorflags/*')
         self.ior_test_sequence = self.params.get(
             "ior_test_sequence", '/run/ior/iorflags/*')
-        self.ior_dfs_oclass = self.params.get(
-            "obj_class", '/run/ior/iorflags/*')
         # Recreate the client hostfile without slots defined
         self.hostfile_clients = write_host_file(
             self.hostlist_clients, self.workdir, None)
 
-    def run_offline_drain_test(self, num_pool, data=False):
+    def run_offline_drain_test(self, num_pool, data=False,
+                               oclass=None, drain_during_aggregation=False):
         """Run the offline drain without data.
             Args:
             num_pool (int) : total pools to create for testing purposes.
             data (bool) : whether pool has no data or to create
                           some data in pool. Defaults to False.
+            oclass (str): DAOS object class (eg: RP_2G1,etc)
+            drain_during_aggregation (bool) : Perform drain and aggregation
+                                              in parallel
         """
         # Create a pool
         pool = {}
```

Comment:
If you will use self.pool = [] and keep appending to this list and use the pools from here, teardown will be able to take care of a lot of cleaning processes and the chances of us leaving behind any unwanted state will be minimized.

Reply:
Presently, I would like to use this as dictionary... The script is basically using to iterative methods and I don't want to touch anything now. Probably on the next PR.

Reply:
Done.